File: Chapter02/Greetings/Greetings/Greetings/GreetingsPage.cs
using System;
using Xamarin.Forms;

namespace Greetings
{
    class GreetingsPage : ContentPage
    {
        public GreetingsPage()
        {
            Content = new Label
            {
                Text = "Greetings, Xamarin.Forms!",
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center
            };
        }
    }
}

File: Chapter02/FS/Greetings/Greetings/GreetingsPage.fs
namespace Greetings

open Xamarin.Forms

//type GreetingsPage() = 
//    inherit ContentPage()
//    let label = new Label()
//    do label.Text <- "Greetings, F# Xamarin.Forms!"
//    do label.VerticalOptions <- LayoutOptions.Center
//    do label.HorizontalOptions <- LayoutOptions.Center
//    do base.Content <- label

// ----- OR -----

//type GreetingsPage() = 
//    inherit ContentPage()
//    let label = Label(Text = "Greetings, F# Xamarin.Forms!",
//                      VerticalOptions = LayoutOptions.Center,
//                      HorizontalOptions = LayoutOptions.Center)
//    do base.Content <- label

// ----- OR -----

type GreetingsPage() = 
    inherit ContentPage(Content = Label(Text = "Greetings, F# Xamarin.Forms!",
                                        VerticalOptions = LayoutOptions.Center,
                                        HorizontalOptions = LayoutOptions.Center))
    